### PR TITLE
Midispeedup

### PR DESCRIPTION
**Pull request by daschuer**
Testing the midi overflow bug, it turns out that Mixxx is not able to hande sysex messages at 3 x Speed. I have read along the code path and found some minor improvements.

This is the successor of https://github.com/mixxxdj/mixxx/pull/826 against master. 

Unfortunately I have no sysex midi controller to provide test data.

### CODE REVIEW

File: src/controllers/controllerengine.cpp [hunk 1]
```diff
@@ -104,9 +104,11 @@ Purpose: Resolves a function name to a QScriptValue including
 Input:   -
 Output:  -
 -------- ------------------------------------------------------ */
-QScriptValue ControllerEngine::resolveFunction(QString function, bool useCache) const {
-    if (useCache && m_scriptValueCache.contains(function)) {
-        return m_scriptValueCache.value(function);
+QScriptValue ControllerEngine::resolveFunction(QString function) const {
```
Comment (Pegasus-RPG) [inline src/controllers/controllerengine.cpp]: Shouldn't this return value be changed to Boolean?

File: src/controllers/controllerengine.cpp [hunk 2]
```diff
@@ -486,21 +483,19 @@ bool ControllerEngine::execute(QString function, const QByteArray data) {
...
-    if (checkException())
-        return false;
-    if (!function.isFunction())
+    if (checkException()) {
         return false;
+    }
 
     QScriptValueList args;
-    args << QScriptValue(m_pBaClass->newInstance(data));
+    args << m_pBaClass->newInstance(data);
```
Comment (Pegasus-RPG) [inline src/controllers/controllerengine.cpp]: How can you remove the QScriptValue cast? Anything passing to script must be a QScriptValue if I recall correctly.

Comment (rryan) [inline src/controllers/controllerengine.cpp]: newInstance returns a QScriptValue

**Comment by rryan** [comment]
lgtm, thanks

**Comment by Pegasus-RPG** [comment]
I need to test this with the SCS.1d.

**Pull request merged**